### DISCUSSION TITLE
Hacking PiVPN: IPv6 prefix delegation

### DISCUSSION THREAD

**Discussion by wargus85** [pivpn/pivpn#1749] [Q&A]
Hi pivpn developers/users!

My ISP allocates an IPv6 /48 subnet. I allocate /64 subnets to my VLANs (via prefix delegation on a Ubiquiti EdgeRouter 4). I would like to allocate a public IPv6 subnet to my PiVPN for use by the wireguard clients - either by setting up a static IPv6 route or automagically allocating the subnet via some IPv6 mechanism.

Is it possible (whats the easiest way) to hack PiVPN such that it will route IPv6 traffic without using NAT, which appears to be the default setup, assuming I can setup static routes between the ER4 and the pivpn virtual machine? I can obviously change the IPv6 subnet that PiVPN uses to my public range such that PiVPN will allocate clients the next IP available in the range, so its really just the nitty gritty of IPv6 persistent rules/wireguard setting I need a little help with. :)

Any suggestions appreciated!

-w

**Comment by andreagdipaolo** [comment]
I'm interested too. Can anyone suggest a solution?
